osc-pcheck.py

```diff
@@ -0,0 +1,123 @@
+# Copyright (C) 2015 SUSE Linux Products GmbH
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+from __future__ import print_function
+from osc import cmdln
+import osc.core
+
+
+@cmdln.option('--push', action='store_true',
+              help="Push changed packages to their parents")
+@cmdln.option('-m', "--message",
+              help='Specify submit message')
+def do_pcheck(self, subcmd, opts, project):
+    """${cmd_name}: Show changed packages (packages that have a diff)
+
+    Examples:
+    osc pcheck <prj>        # shows changed packages etc. for <prj>
+
+    --push      Create submit requests for packages with a diff (if none exists yet)
+    -m          Specify submit message (defaut: "Scripted push of project <prj>")
+
+    """
+    apiurl = self.get_api_url()
+    sinfos = osc.core.get_project_sourceinfo(apiurl, project, True)
+    todo = {}
+    errors = {}
+    md5s = {}
+    pmap = {}
+    changed = []
+    changeSRed = {}
+    api = oscapi(apiurl)
+    for pkg, sinfo in sinfos.items():
+        if sinfo.find('error'):
+            errors[pkg] = sinfo.find('error').text
+            continue
+        elif sinfo.find('originpackage') is not None:
+            # This is a package created from a _multibuild
+            # Status will be checked by the main one (which
+            # has no originpackage.) so let's not continue further
+            continue
+        elif sinfo.find('linked') is not None:
+            elm = sinfo.find('linked')
+            key = '%s/%s' % (elm.get('project'), elm.get('package'))
+            pmap.setdefault(key, []).append(pkg)
+            todo.setdefault(elm.get('project'), []).append(elm.get('package'))
+        md5s[pkg] = sinfo.get('verifymd5')
+    for prj, pkgs in todo.items():
+        sinfos = osc.core.get_project_sourceinfo(apiurl, prj, True, *pkgs)
+        for pkg, sinfo in sinfos.items():
+            key = '%s/%s' % (prj, pkg)
+            for p in pmap[key]:
+                vmd5 = md5s.pop(p)
+                if vmd5 != sinfo.get('verifymd5'):
```

Comment:
that's a little heavy indented. Just negate the != here and continue, so you can move the following block one left. Or possibly use a function